src/write_gcm.c

```diff
@@ -3568,15 +3568,20 @@ static void *wg_process_queue(wg_context_t *ctx, wg_queue_t *queue,
     }
     if (wg_update_stats(stats) != 0) {
       wg_some_error_occured_g = 1;
-      WARNING("%s: wg_update_stats failed.", this_plugin_name);
+      ERROR("%s: wg_update_stats failed.", this_plugin_name);
       break;
     }
     payloads = NULL;
   }
 
  leave:
   wg_deriv_tree_destroy(deriv_tree);
-  WARNING("write_gcm: Consumer thread is exiting.");
+  if (!queue->request_terminate) {
+    queue->consumer_thread_created = 0;
```

Comment:
Marking as resolved given as we agreed that complete shutdown is preferable behavior than leaving the agent in a contradictory state.